packages/twenty-server/src/engine/core-modules/billing/services/billing-meter-event.service.ts

```diff
@@ -0,0 +1,95 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { NonNegative } from 'type-fest';
+import { Repository } from 'typeorm';
+
+import {
+  BillingException,
+  BillingExceptionCode,
+} from 'src/engine/core-modules/billing/billing.exception';
+import { BillingCustomer } from 'src/engine/core-modules/billing/entities/billing-customer.entity';
+import { BillingMeterEventName } from 'src/engine/core-modules/billing/enums/billing-meter-event-names';
+import { BillingSubscriptionService } from 'src/engine/core-modules/billing/services/billing-subscription.service';
+import { StripeBillingMeterEventService } from 'src/engine/core-modules/billing/stripe/services/stripe-billing-meter-event.service';
+import { FeatureFlagKey } from 'src/engine/core-modules/feature-flag/enums/feature-flag-key.enum';
+import { FeatureFlagService } from 'src/engine/core-modules/feature-flag/services/feature-flag.service';
+
+@Injectable()
+export class BillingMeterEventService {
+  protected readonly logger = new Logger(BillingMeterEventService.name);
+  constructor(
+    private readonly stripeBillingMeterEventService: StripeBillingMeterEventService,
+    @InjectRepository(BillingCustomer, 'core')
+    private readonly billingCustomerRepository: Repository<BillingCustomer>,
+    private readonly featureFlagService: FeatureFlagService,
+    private readonly billingSubscriptionService: BillingSubscriptionService,
+  ) {}
+
+  async hasActiveBillingSubscription(workspaceId: string): Promise<boolean> {
+    const activeWorkspaceSubscription =
+      await this.billingSubscriptionService.getCurrentBillingSubscriptionOrThrow(
+        {
+          workspaceId,
+        },
+      );
+
+    return !!activeWorkspaceSubscription;
+  }
+
+  async sendBillingMeterEvent({
```

Comment:
`event` is misleading with the events system in twenty-server. I suggest this name
```suggestion
  async billUsage({
```